stdlib/public/core/IntegerParsing.swift

```diff
@@ -10,15 +10,404 @@
+  /// (`+` or `-`), followed by one or more numeric digits (`0-9`) or letters
+  /// (`a-z` or `A-Z`). Parsing of the string is case insensitive.
+  ///
+  ///     let x = Int("123")
+  ///     // x == 123
+  ///
+  ///     let y = Int("-123", radix: 8)
+  ///     // y == -83
+  ///     let y = Int("+123", radix: 8)
+  ///     // y == +83
+  ///
+  ///     let z = Int("07b", radix: 16)
+  ///     // z == 123
+  ///
+  /// If `text` is in an invalid format or contains characters that are out of
+  /// bounds for the given `radix`, or if the value it denotes in the given
+  /// `radix` is not representable, the result is `nil`. For example, the
+  /// following conversions result in `nil`:
+  ///
+  ///     Int(" 100")                       // Includes whitespace
+  ///     Int("21-50")                      // Invalid format
+  ///     Int("ff6600")                     // Characters out of bounds
+  ///     Int("zzzzzzzzzzzzz", radix: 36)   // Out of range
+  ///
+  /// - Parameters:
+  ///   - text: The ASCII representation of a number in the radix passed as
+  ///     `radix`.
+  ///   - radix: The radix, or base, to use for converting `text` to an integer
+  ///     value. `radix` must be in the range `2...36`. The default is 10.
+  @inlinable @inline(__always)
+  public init?<S : StringProtocol>(_ text: S, radix: Int = 10) {
+    precondition(2...36 ~= radix)
+    let result_: Self?
+    if let text = text as? String, text._guts.isSmall {
+      result_ = _parseSmall(text._guts.rawBits, radix: radix)
+    } else {
+      result_ = text.utf8.withContiguousStorageIfAvailable { utf8 -> Self? in
+        var iter = utf8.makeIterator()
+        return _parseFromUTF8Inlined(&iter, radix: radix)
+      } ?? {
+        var iter = text.utf8.makeIterator()
+        return _parseFromUTF8(&iter, radix: radix)
+      }()
+    }
+    guard _fastPath(result_ != nil), let result = result_ else { return nil }
+    self = result
+  }
+
+  /// Creates a new integer value from the given string.
+  ///
+  /// The string passed as `description` may begin with a plus or minus sign
+  /// character (`+` or `-`), followed by one or more numeric digits (`0-9`).
+  ///
+  ///     let x = Int("123")
+  ///     // x == 123
+  ///
+  /// If `description` is in an invalid format, or if the value it denotes in
+  /// base 10 is not representable, the result is `nil`. For example, the
+  /// following conversions result in `nil`:
+  ///
+  ///     Int(" 100")                       // Includes whitespace
+  ///     Int("21-50")                      // Invalid format
+  ///     Int("ff6600")                     // Characters out of bounds
+  ///     Int("10000000000000000000000000") // Out of range
+  ///
+  /// - Parameter description: The ASCII representation of a number.
+  @inlinable
+  public init?(_ description: String) {
+    self.init(description, radix: 10)
+  }
+}
+
+@inlinable @inline(__always)
```

Comment:
See above.  It would be nice to avoid inlining functions at this level.

Reply:
This is definitely way too much inlining; but without inlining the generics don't go away. The fast-paths we care about are mostly around contiguous UTF-8 Strings/Substrings being parsed into results that fit in 64-bits. The special path for small strings might be warranted. However, we should be able to strip all generics fairly early on for fast paths and invoke non-inlined code.